legal-api/src/legal_api/models/batch_processing.py

```diff
@@ -0,0 +1,93 @@
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""This module holds data for batch processing."""
+from enum import auto
+
+from legal_api.utils.base import BaseEnum
+from legal_api.utils.datetime import datetime
+
+from .db import db
+
+
+class BatchProcessing(db.Model):  # pylint: disable=too-many-instance-attributes
+    """This class manages the batch processing."""
+
+    class BatchProcessingStep(BaseEnum):
+        """Render an Enum of the batch processing step."""
+
+        WARNING_LEVEL_1 = auto()
+        WARNING_LEVEL_2 = auto()
+        DISSOLUTION = auto()
+
+    class BatchProcessingStatus(BaseEnum):
+        """Render an Enum of the batch processing status."""
+
+        HOLD = auto()
+        PROCESSING = auto()
+        WITHDRAWN = auto()
+        COMPLETED = auto()
+        ERROR = auto()
+
+    __tablename__ = 'batch_processing'
+
+    id = db.Column(db.Integer, primary_key=True)
+    business_identifier = db.Column('business_identifier', db.String(10), default='', nullable=False)
+    step = db.Column('step', db.Enum(BatchProcessingStep), nullable=False)
+    status = db.Column('status', db.Enum(BatchProcessingStatus), nullable=False)
+    notes = db.Column('notes', db.String(150), default='', nullable=True)
+    created_date = db.Column('created_date', db.DateTime(timezone=True), default=datetime.utcnow)
+    last_modified = db.Column('last_modified', db.DateTime(timezone=True), default=datetime.utcnow)
+
+    # parent keys
+    batch_id = db.Column('batch_id', db.Integer, db.ForeignKey('batches.id'), index=True, nullable=False)
+    business_id = db.Column('business_id', db.Integer, db.ForeignKey('businesses.id'), index=True, nullable=False)
+
+    def save(self):
+        """Save the object to the database immediately."""
+        db.session.add(self)
+        db.session.commit()
+
+    @classmethod
+    def find_by_id(cls, batch_processing_id: int):
+        """Return the batch matching the id."""
+        batch_processing = None
+        if batch_processing_id:
+            batch_processing = cls.query.filter_by(id=batch_processing_id).one_or_none()
+        return batch_processing
+
+    @classmethod
+    def find_by(cls,  # pylint: disable=too-many-arguments
+                batch_id: int = None,
+                business_id: int = None,
+                step: BatchProcessingStep = None,
+                status: BatchProcessingStatus = None,
+                notes: str = None) -> dict:
```

Comment:
can we remove the `notes` parameter too?

Reply:
removed